frontend/ui/impl/src/main/java/com/virtuslab/gitmachete/frontend/ui/impl/table/EnhancedGraphTable.java

```diff
@@ -400,7 +497,11 @@ public void queueRepositoryUpdateAndModelRefresh(@UI Runnable doOnUIThreadWhenRe
         setTextForEmptyTable(getString("string.GitMachete.EnhancedGraphTable.empty-table-text.loading"));
 
         LOG.debug("Queuing repository update onto a non-UI thread");
-        new GitMacheteRepositoryUpdateBackgroundable(project, gitRepository, branchLayoutReader, doRefreshModel).queue();
+        new GitMacheteRepositoryUpdateBackgroundable(project,
+            gitRepository,
+            branchLayoutReader,
+            doRefreshModel,
+            /* gitMacheteRepositoryConsumer */ repository -> gitMacheteRepository = repository).queue();
```

Comment:
For some reason this is not captured by UI Effect Checker... but `gitMacheteRepository` won't be assigned on UI thread right? 🤔 if these assignments outside of UI thread are unavoidable, consider using `AtomicReference`